homeassistant/components/ping/binary_sensor.py

```diff
@@ -145,7 +148,7 @@ def ping(self):
         except (subprocess.CalledProcessError, AttributeError):
```

Comment:
Please check what exceptions we need to catch now.

Reply:
Bump.

Reply:
@Misiu 

Once #38504 is merged, I'll take a look at this.

Would you add code to implement a timeout like #38491 so when we come back and replace #38504 with this code we don't regress that issue

Reply:
@bdraco sure thing. I'll do that when I return from the vacations.
I'll be grateful for your review and help with finishing this. Then ping device tracker will be next!

Reply:
@bdraco I've added a timeout to my PR.
I've also tried to add more types to this class, so I'll be grateful if you could look at that too.